pineappl_capi/src/lib.rs

```diff
@@ -1357,3 +1355,253 @@ pub unsafe extern "C" fn pineappl_string_delete(string: *mut c_char) {
+    orders: usize,
+    order_params: *const u32,
+    bins: usize,
+    bin_limits: *const f64,
+    nb_convolutions: usize,
+    convolution_types: *const *const c_char, // takes: `const char* convtype[] = { "UnpolPDF" };`
+    pdg_ids: *const c_int,                   // takes: `const char* pdg_dis[] = { 2212 }`
+    kinematics: *const *const c_char, // takes: `const char* kinematics[] = { "scale", "x1" }`
+    interpolations: *const InterpTuples, // takes: `struct InterpTuples interpdata[] = { {...} }`
+) -> Box<Grid> {
+    // PID Basis
+    let pid_basis = match unsafe {
+        CStr::from_ptr(pid_basis)
+            .to_str()
+            .expect("Failed to convert `pid_basis` into a proper string.")
+    } {
+        "Evol" => PidBasis::Evol,
+        "Pdg" => PidBasis::Pdg,
+        _ => todo!(),
+    };
+
+    // Luminosity channels
+    let channels = unsafe { &*channels };
+
+    // Perturbative orders
+    let order_params = unsafe { slice::from_raw_parts(order_params, 4 * orders) };
+    let orders: Vec<_> = order_params
+        .chunks(4)
+        .map(|s| Order {
+            alphas: s[0].try_into().unwrap(),
+            alpha: s[1].try_into().unwrap(),
+            logxir: s[2].try_into().unwrap(),
+            logxif: s[3].try_into().unwrap(),
+            logxia: s[4].try_into().unwrap(),
+        })
+        .collect();
+
+    // Bin limits
+    let bin_limits = unsafe { slice::from_raw_parts(bin_limits, bins + 1).to_vec() };
+
+    // Types of convolutions
+    let mut convolutions = Vec::new();
+    for i in 0..nb_convolutions {
+        let c_conv = unsafe {
+            let str_ptr = *convolution_types.add(i);
+            CStr::from_ptr(str_ptr)
+                .to_str()
+                .expect("Failed to convert one of the `convolution_types` into a proper string.")
+        };
+
+        let pid_value = unsafe { *pdg_ids.add(i) };
+
+        let cvtype = match c_conv {
+            "UnpolPDF" => Conv::new(ConvType::UnpolPDF, pid_value),
+            "PolPDF" => Conv::new(ConvType::PolPDF, pid_value),
+            "UnpolFF" => Conv::new(ConvType::UnpolFF, pid_value),
+            "PolFF" => Conv::new(ConvType::PolFF, pid_value),
+            _ => todo!(),
+        };
+        convolutions.push(cvtype);
+    }
+
+    // Grid interpolations
+    let mut interp_vecs = Vec::new();
+    let interp_slices = unsafe { std::slice::from_raw_parts(interpolations, nb_convolutions + 1) };
+    for interp_ref in interp_slices {
+        interp_vecs.push(construct_interpolation(interp_ref));
+    }
+
+    // Kinematics
+    let mut kinematics_vec = Vec::new();
+    for k in 0..=nb_convolutions {
+        let kin = unsafe {
+            let kin_ptr = *kinematics.add(k);
+            CStr::from_ptr(kin_ptr)
+                .to_str()
+                .expect("Failed to convert one of the `kinematics` types into a proper string.")
+        };
+
+        let kin_type = match kin {
+            "scale" => Kinematics::Scale(0),
```

Comment:
This we must make more general, because we'll also need at least `Kinematics::Scale(1)` for flexible-scale fastNLO tables.

Reply:
Thanks for the comments @cschwan! I will address them.

As for this particular one, it is already superseded by this commit: [5ff86af](https://github.com/NNPDF/pineappl/pull/299/commits/5ff86af1d6c649336ed4cbf44c1badb591dfe6ee) which indeed makes it more general.

Reply:
For `x` yes, but I don't see how `Scale(1)` can be constructed.